### PR TITLE
Suppress CA1860 when in expression tree

### LINKED ISSUE #7063: Allow us to configure CA1860 so that it does not apply to expression trees
Hi, I have an issue with [CA1860: Avoid using 'Enumerable.Any()' extension method](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1860).

Using `.Any()` over `.Count` is almost always desirable when working with expression trees and Entity Framework, because EF converts those expression trees into SQL queries. The query that EF generates under the hood is much more performant when using `.Any()` (tested with EF6). This makes sense, because in SQL, doing a `COUNT()` just to check if a (sub)query is not empty is not the optimal way to achieve this. When using `.Any()`, EF correctly generates a SQL query that uses `EXISTS`.

It would be great if it were possible to configure CA1860 so that it ignores expression trees. For example, something like this:

```
# example example example
dotnet_diagnostic.CA1860.ignore_expression_trees = true
```

### PR DESCRIPTION
**Pull request by CollinAlpert**
Fixes #7063

### CODE REVIEW

File: src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Performance/PreferLengthOverAnyTests.cs [hunk 1]
```diff
@@ -254,5 +255,70 @@ public class MyCollection {
...
+                                    public void M() {
+                                        var array = new int[0];
+                                        Evaluate(() => array.Any());
+                                    }
+                                
+                                    private void Evaluate(Expression<Func<bool>> expression)
+                                    {
+                                    }
+                                }
+                                """;
```
Comment (sharwell) [inline src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Performance/PreferLengthOverAnyTests.cs:278]: 💡 The indentation here would be vastly improved if the closing `"""` was indented exactly 4 spaces relative to `const`

**Review by sharwell — APPROVED** [review]

**Review by mavasani — APPROVED** [review]

File: src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Performance/PreferLengthOverAnyTests.cs (hunk 1 above)
Comment (CollinAlpert) [reply src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Performance/PreferLengthOverAnyTests.cs]: Like this?

**Comment by ArturDorochowicz** [comment]
@mavasani When is it going to be released? Before .net 9?

**Comment by CollinAlpert** [comment]
@ArturDorochowicz You can reference [this package version](https://www.nuget.org/packages/Microsoft.CodeAnalysis.NetAnalyzers/9.0.0-preview.24122.1) to get the fix.

**Pull request merged**